File: app_vlxd/lib/model/VatTu.dart
import 'package:app_vlxd/model/LoaiVT.dart';

class VatTu {
  final String ma;
  final String ten;
  final String dvt;
  final double donGia;
  final String image;
  final LoaiVT loaiVT;

  VatTu({
    required this.ma,
    required this.ten,
    required this.dvt,
    required this.donGia,
    required this.image,
    required this.loaiVT,
  });

  factory VatTu.fromJson(Map<String, dynamic> json) {
    return VatTu(
      ma: json['ma'],
      ten: json['ten'],
      dvt: json['dvt'],
      donGia: json['donGia'],
      image: json['image'],
      loaiVT: LoaiVT.fromJson(json['loaivt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'ma': ma,
      'ten': ten,
      'dvt': dvt,
      'donGia': donGia,
      'image': image,
      'loaivt': loaiVT.toJson(),
    };
  }
}

File: vatlieuxaydung/src/main/java/com/example/vatlieuxaydung/model/VatTu.java
package com.example.vatlieuxaydung.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

@Entity(name = "vattu")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class VatTu {
    @Id
    @Column(name = "Ma", columnDefinition = "CHAR(30)")
    String ma;
    @Column(name = "Ten")
    String ten;
    @Column(name = "DVT")
    String dvt;
    @Column(name = "dongia")
    Double donGia;
    String image;
    @ManyToOne
    @JoinColumn(name="LoaiVT")
    @JsonIgnoreProperties("dsVatTu")
    LoaiVT loaivt;

}
